content/sensu-go/6.2/operations/monitoring-as-code/_index.md

```diff
@@ -0,0 +1,151 @@
+If you manage your monitoring and observability workflow in the same way as the systems you're monitoring &mdash; as code, using a centralized continuous integration/continuous delivery (CI/CD) pipeline &mdash; you can align your monitoring with your product and services and improve visibility, reliability, and repeatability.
+
+- Your entire team can share, edit, review, and version Sensu configuration files just like they would with other "as-code" solutions.
+- Export your Sensu configuration from one environment and initialize another environment with the same configuration.
+- Use Sensu configuration files to remove, restore, back up, and recover Sensu instances.
+
+## Create a monitoring as code repository
+
+You can use any source control repository for maintaining your monitoring workflow configuration files.
+Your monitoring as code repository should include configuration files with the resource definitions you use in your monitoring workflow.
+
+Use `sensuctl dump` to [export some or all of your existing Sensu resource definitions][7] to standard out (STDOUT) or to a file.
+
+You can also use sensuctl commands to retrieve your Sensu resource definitions as you create them so you can copy them into your configuration files.
+For example, to retrieve the definition for a check named `check-cpu`:
+
+{{< language-toggle >}}
+
+{{< code shell "YML" >}}
+sensuctl check info check-cpu --format yaml
+type: CheckConfig
+api_version: core/v2
+metadata:
+  created_by: admin
+  name: check-cpu
+  namespace: default
+spec:
+  check_hooks: null
+  command: check-cpu.rb -w 75 -c 90
+  env_vars: null
+  handlers:
+  - slack
+  high_flap_threshold: 0
+  interval: 60
+  low_flap_threshold: 0
+  output_metric_format: ""
+  output_metric_handlers: null
+  proxy_entity_name: ""
+  publish: true
+  round_robin: false
+  runtime_assets:
+  - cpu-checks-plugins
+  - sensu-ruby-runtime
+  secrets: null
+  stdin: false
+  subdue: null
+  subscriptions:
+  - system
+  timeout: 0
+  ttl: 0
+{{< /code >}}
+
+{{< code shell "JSON" >}}
+sensuctl check info check-cpu --format json
+{
+  "command": "check-cpu.rb -w 75 -c 90",
+  "handlers": [
+    "slack"
+  ],
+  "high_flap_threshold": 0,
+  "interval": 60,
+  "low_flap_threshold": 0,
+  "publish": true,
+  "runtime_assets": [
+    "cpu-checks-plugins",
+    "sensu-ruby-runtime"
+  ],
+  "subscriptions": [
+    "system"
+  ],
+  "proxy_entity_name": "",
+  "check_hooks": null,
+  "stdin": false,
+  "subdue": null,
+  "ttl": 0,
+  "timeout": 0,
+  "round_robin": false,
+  "output_metric_format": "",
+  "output_metric_handlers": null,
+  "env_vars": null,
+  "metadata": {
+    "name": "check-cpu",
+    "namespace": "default",
+    "created_by": "admin"
+  },
+  "secrets": null
+}
+{{< /code >}}
+
+{{< /language-toggle >}}
+
+Many of our [guides][6] demonstrate how to create, update, and retrieve resource definitions in this way.
```

Comment:
They *will* include this instruction, although they mostly don't right now. See https://github.com/sensu/sensu-docs/pull/2962 -- we think the monitoring as code project should include reworking pages that use mainly sensuctl commands to create resources so that they instead focus on complete code examples.

I also think we need to expand this explanation so it's clear to users what they should do with the resource definitions once they get them with `sensuctl [resource] info [resource name]` -- for example, save them in a file like `checks.yaml`?

Reply:
To capture resource to a new file (or overwrite existing), e.g. resource.yml:
`sensuctl [resource] info [resource name] > resource.yml`

To capture resource to a new file (or overwrite existing), e.g. resource.yml and see output on standard out:
`sensuctl [resource] info [resource name] | tee resource.yml`

To append resource to an existing file:
`sensuctl [resource] info [resource name] >> resource.yml`

To append resource to an existing file and show the output on standard out:
`sensuctl [resource] info [resource name] | tee -a resource.yml`